File: Solutions/216. Combination Sum III/216. Combination Sum III.go
func combinationSum3(k int, n int) [][]int {
    var res [][]int
    var tmp[]int
    combination(&res,tmp,k,1,n)
    return res
}

func combination(res *[][]int,tmp []int,k,st,n int) {
    if len(tmp)==k && n==0{
        x:=make([]int,len(tmp))
        copy(x,tmp)
        *res=append(*res,x)
        return;
    }
    for i:=st;i<=9;i+=1{
        tmp=append(tmp,i)
        combination(res,tmp,k,i+1,n-i)
        tmp=tmp[:len(tmp)-1]
    }
}

File: Solutions/216. Combination Sum III/216. Combination Sum III.cpp
class Solution {
public:
    vector<vector<int>> combinationSum3(int k, int n) {
        vector<vector<int>> res;
        vector<int> tmp;
        combination(res,tmp,k,1,n);
        return res;
    }
    void combination(vector<vector<int>>& res,vector<int>& tmp,int& k,int st,int n){
        if(tmp.size()==k and n==0){
            res.push_back(tmp);
            return;
        }
        for(int i=st;i<=9;++i){
            tmp.push_back(i);
            combination(res,tmp,k,i+1,n-i);
            tmp.pop_back();
        }
    }
};